### PR TITLE
[11.x] Improves errors

### PR DESCRIPTION
**Pull request by nunomaduro**

### CODE REVIEW

File: src/Illuminate/Foundation/resources/exceptions/renderer/components/card.blade.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<section
+    {{ $attributes->merge(['class' => "@container flex flex-col p-6 sm:p-12 bg-white dark:bg-gray-900/80 text-gray-900 dark:text-gray-100 rounded-lg default:col-span-full default:lg:col-span-6 default:row-span-1 dark:ring-1 dark:ring-gray-800 shadow-xl"]) }}"
```
Comment (devajmeireles) [inline src/Illuminate/Foundation/resources/exceptions/renderer/components/card.blade.php:2]: `@container` ➡️  https://github.com/laravel/framework/issues/50171

File: src/Illuminate/Foundation/Exceptions/Renderer/Frame.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+class Frame
+{
+    use ResolvesDumpSource;
+
+    /**
+     * The "flatten" exception instance.
+     *
+     * @var \Symfony\Component\ErrorHandler\Exception\FlattenException
+     */
+    protected $exception;
```
Comment (marius-mcp) [inline src/Illuminate/Foundation/Exceptions/Renderer/Frame.php:17]: If this is a new class, why not declare inline the type for properties?

Comment (marius-mcp) [reply src/Illuminate/Foundation/Exceptions/Renderer/Frame.php:17]: Analog for return types.

File: src/Illuminate/Foundation/Exceptions/Renderer/Listener.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     *
+     * @param  \Illuminate\Contracts\Events\Dispatcher  $events
+     * @return void
+     */
+    public function registerListeners(Dispatcher $events)
+    {
+        $events->listen(QueryExecuted::class, [$this, 'onQueryExecuted']);
+
+        if (isset($_SERVER['LARAVEL_OCTANE'])) {
+            $events->listen([RequestReceived::class, TaskReceived::class, TickReceived::class, RequestTerminated::class], function () {
```
Comment (taylorotwell) [inline src/Illuminate/Foundation/Exceptions/Renderer/Listener.php:32]: @nunomaduro does this need to also reset on any job events?

**Comment by taylorotwell** [comment]
@nunomaduro I think this is good pending those weird HTML formatting things I pointed out and single comment above.

File: src/Illuminate/Foundation/Providers/FoundationServiceProvider.php [hunk 4]
```diff
@@ -198,6 +210,36 @@ protected function registerExceptionTracking()
         });
     }
 
+    /**
+     * Register the exceptions renderer.
+     *
+     * @return void
+     */
+    protected function regsiterExceptionRenderer()
```
Comment (PerryvanderMeer) [inline src/Illuminate/Foundation/Providers/FoundationServiceProvider.php:218]: Regsiter ➡️ Register

File: src/Illuminate/Foundation/Exceptions/Renderer/Listener.php (hunk 3 above)
Comment (nunomaduro) [reply src/Illuminate/Foundation/Exceptions/Renderer/Listener.php:32]: I've replicated what ignition is doing: https://github.com/spatie/laravel-ignition/blob/9c60d929a91803dee7e47614c37cbe6d3f2b1e2c/src/IgnitionServiceProvider.php#L311C1-L311C43.

**Comment by driesvints** [comment]
@mfn gonna be super hard to deduct what's going on without more info. Best that you attempt a minimum repro example with the installer:

```bash
laravel new bug-report --github="--public"
```

**Comment by driesvints** [comment]
I'm sorry @mfn but we can't go off on a third party repo. We'll need a clear way to reproduce this in a Laravel project.

**Comment by NickSdot** [comment]
@nunomaduro I am missing two functionalities I pretty much appreciated before:

1) Error lines where clickable to open in editor
2) Output of dd() and dump() were visible

Before I am going to dig: should these features work or were thet left out intantionaly?

**Comment by AhmedAlaa4611** [comment]
@nunomaduro I think this now should be updated to `tailwind v4`

**Comment by str** [comment]
+1 for "open in editor"  feature

**Comment by NickSdot** [comment]
> +1 for "open in editor"  feature

`spatie/laravel-ignition`

**Comment by str** [comment]
I found it!  

You need to define `'editor' => env('APP_EDITOR'),` in your `app.php`

**Pull request merged**